src/library/scala/collection/immutable/Stream.scala

```diff
@@ -371,13 +371,21 @@ object Stream extends SeqFactory[Stream] {
 
   @SerialVersionUID(3L)
   final class Cons[A](override val head: A, tl: => Stream[A]) extends Stream[A] {
```

Comment:
would `private[this] var tl: () => Stream[A]` be binary-compatible, since by-name values are compiled to `Function0`s?

Reply:
I suppose so, but the class is public and it wouldn't be source compatible.